velox/docs/stats.rst

```diff
@@ -0,0 +1,122 @@
+==============
+Runtime Metric
+==============
+
+Runtime metric is used to collect the metrics of important velox runtime events
+for monitoring purpose. The collected metrics can provide insights into the
+continuous availability and performance analysis of a velox runtime system. For
+instance, the collected data can help automatically generate alerts at an
+outage. Velox provides a framework to collect the metrics which consists of
+three steps:
+
+**Define**: define the name and type for the metric through DEFINE_METRIC and
+DEFINE_HISTOGRAM_METRIC macros. DEFINE_HISTOGRAM_METRIC is used for histogram
+metric type and DEFINE_METRIC is used for the other types (see metric type
+definition below). BaseStatsReporter provides methods for metric definition.
+Register metrics during startup using registerVeloxMetrics() API.
+
+**Record**: record the metric data point using RECORD_METRIC_VALUE and
+RECORD_HISTOGRAM_METRIC_VALUE macros when the corresponding event happens in
+production. RECORD_HISTOGRAM_METRIC_VALUE is used for histogram metric type and
+RECORD_METRIC_VALUE is used for the other types. BaseStatsReporter provides
+methods for metric recording.
+
+**Export**: aggregates the collected data points based on the defined metrics,
+and periodically exports to the backend monitoring service, such as ODS used by
+Meta, OpenCensus and Prometheus provided by OSS. A derived implementation of
```

Comment:
"provided by OSS" - perhaps, provide links to corresponding projects and mention whether they are Apache projects or something else.

Reply:
Added the project link. Thanks!